source/interface/hud/UnitRecruitmentCard.gd

```diff
@@ -9,7 +9,7 @@ var unit_type_id: String
 
 func initialize(unit_type_id: String) -> void:
 	self.unit_type_id = unit_type_id
-	var unit_type: PackedScene = Registry.units[unit_type_id].instance()
+	var unit_type: Node2D = Registry.units[unit_type_id].instance()
```

Comment:
I think you can use `UnitType` here as class instead of Node2D
if it `var unit_type := Registry.units[unit_type_id].instance() as UnitType` like that.

Reply:
thanks, I did 
`var unit_type: UnitType = Registry.units[unit_type_id].instance()`
is it any different from 
`var unit_type := Registry.units[unit_type_id].instance() as UnitType` ?

Reply:
slightly.
`var unit_type: UnitType = Registry.units[unit_type_id].instance()`
instance() returns a `Node` type object that then is assigned to a variable of type `UnitType`
I'm not entirely sure but I think this will give you an error as you can't save a Node into a UnitType variable.

with `var unit_type := Registry.units[unit_type_id].instance() as UnitType`
you explicitly cast the value of instance() to a UnitType, before assigning it to the variable.

Reply:
I changed to your version however:

> instance() returns a Node type 
I believe that is not correct. `instance()` returns in fact `UnitType`, but it is only know at runtime - but we know it from the code logic. Therefore cast `as UnitType` is redundant as variable is already of that type and is used only to deduce this type via `:=` operator, which is the same as simply declaring variable of that type (it gets casted implicitly anyways). I wouldn't even be surprised if GDScript parser produced same output in both cases.

> but I think this will give you an error
I'm testing my stuff before submitting PR ;) Both methods work.